docs/lang/articles/basic/type.md

```diff
@@ -302,16 +84,30 @@ def func(a: ti.f32) -> ti.i64:
     ...
 ```
 
-### Type casts
+### Type promotion
 
-All data types are static in the **Taichi scope**. Therefore, casts are needed when you want to assign a certain type of data to another one.
+When Taichi performs binary operations on different types, the result is a
+promoted type. This is known as type promotion. Following the C programming
+language convention, Taichi always chooses the more precise type to contain the
+result value. For example:
 
-#### Implicit casts
+- `i32 + f32 = f32` (integer + float = float)
+- `i32 + i64 = i64` (less-bits + more-bits = more-bits)
+
+### Type casts
+
+When using Taichi, you may encounter situations when certain type of data needs
+to be assigned to another type. In this case, type casts are needed. Here, we
+list four use cases commonly seen during Taichi programming, namely implicit
+casts, explicit casts, casting vectors and matrices, and bit-casts.
 
 :::caution
-The type of a variable is **determined on its initialization**.
+In Taichi, the type of a variable is **determined on its initialization**.
+All data types are static in the **Taichi scope**.
```

Comment:
```suggestion
All data types are static in the **Taichi scope**. (A static type of variable means the type is known at compile time.  This lets the compiler decide whether a given variable can perform the actions requested from it or not. For more information, see [the difference between statically typed and dynamically typed languages](https://stackoverflow.com/questions/1517582/what-is-the-difference-between-statically-typed-and-dynamically-typed-languages)
```

Reply:
@mzmzm thanks, can you adapt the sentence a bit?
1: (Static here means...... Based on this information, the compiler can decide whether the performed action is legal or not.). 
2: please don't use stackoverflow as an official reference in our document. Wikipedia is fine. cppreference.com is fine. but not random post. Can you try to find another good reference to be linked?

Reply:
Thanks. I update my suggestion.

Reply:
Did you forget to apply suggestions here @qiao-bo ?